controllers/storagecluster/storageconsumer.go

```diff
@@ -136,15 +136,36 @@ func (s *storageConsumer) ensureCreated(r *StorageClusterReconciler, storageClus
 			availableServices,
 		)
 		resourceMap := util.WrapStorageConsumerResourceMap(data)
-		if backwardCompatibleInfo.Pre4_19InternalConsumer == "" {
+		if backwardCompatibleInfo.Pre4_19DeploymentMode == "" {
```

Comment:
This is wrong, if you are not in backward compatibility mode then you should not even look at this struct.